### PR TITLE
Sync labs Core functionality

### PR DESCRIPTION
**Pull request by IanTeo**
Add Infrastructure and basic functionality for a new Lab: Sync Labs

The PR is split into 2 parts,
1. The infrastructure done by @Kenneth-LJS which is under the commit "Add Sync Labs Infrastucture"
1. The basic functionality done by myself which is under the commit "Add Sync labs basic functionality"

This is the squashed version of #1174 which has my part included.
@damithc I hope documenting it like this makes it clear for you to see who has done what. Future improvements will not have the problem of overlapping PRs.

### CODE REVIEW

File: PowerPointLabs/PowerPointLabs/SyncLab/SyncFormatUtil.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    {
+
+        public static void SyncColorFormat(ColorFormat formatToApplyOn, ColorFormat formatToApply)
+        {
+            try
+            {
+                formatToApplyOn.ObjectThemeColor = formatToApply.ObjectThemeColor;
+            }
+            catch (ArgumentException)
+            {
```
Comment (initialshl) [inline PowerPointLabs/PowerPointLabs/SyncLab/SyncFormatUtil.cs]: Maybe log the exception?

File: PowerPointLabs/PowerPointLabs/SyncLab/SyncFormatUtil.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            }
+            try
+            {
+                formatToApplyOn.SchemeColor = formatToApply.SchemeColor;
+            }
+            catch (ArgumentException)
+            {
```
Comment (initialshl) [inline PowerPointLabs/PowerPointLabs/SyncLab/SyncFormatUtil.cs]: Log exception here too

File: PowerPointLabs/PowerPointLabs/SyncLab/SyncFormatUtil.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                formatToApplyOn.BeginArrowheadLength = formatToApply.BeginArrowheadLength;
+                formatToApplyOn.BeginArrowheadStyle = formatToApply.BeginArrowheadStyle;
+                formatToApplyOn.BeginArrowheadWidth = formatToApply.BeginArrowheadWidth;
+                formatToApplyOn.EndArrowheadLength = formatToApply.EndArrowheadLength;
+                formatToApplyOn.EndArrowheadStyle = formatToApply.EndArrowheadStyle;
+                formatToApplyOn.EndArrowheadWidth = formatToApply.EndArrowheadWidth;
+            }
+            catch (ArgumentException)
+            {
```
Comment (initialshl) [inline PowerPointLabs/PowerPointLabs/SyncLab/SyncFormatUtil.cs]: Log this too

File: PowerPointLabs/PowerPointLabs/SyncLab/SyncFormatUtil.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            }
+            formatToApplyOn.DashStyle = formatToApply.DashStyle;
+            SyncColorFormat(formatToApplyOn.ForeColor, formatToApply.ForeColor);
+            formatToApplyOn.InsetPen = formatToApply.InsetPen;
+            try
+            {
+                formatToApplyOn.Pattern = formatToApply.Pattern;
+            }
+            catch (ArgumentException)
+            {
```
Comment (initialshl) [inline PowerPointLabs/PowerPointLabs/SyncLab/SyncFormatUtil.cs]: Log exception

File: PowerPointLabs/PowerPointLabs/SyncLab/View/SyncFormatDialog.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                        {
+                            children.Add((SyncFormatDialogItem)treeItem);
+                        }
+                        else if (treeItem is TreeViewItem)
+                        {
+                            children.Add((SyncFormatDialogItem)((TreeViewItem)treeItem).Header);
+                        }
+                        else
+                        {
+                            throw new Exception("");
```
Comment (initialshl) [inline PowerPointLabs/PowerPointLabs/SyncLab/View/SyncFormatDialog.xaml.cs]: Use descriptive exception with message too

**Comment by Kenneth-LJS** [comment]
Will do exception logging in next iteration

**Pull request merged**